src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const returnedObject: GroupsMap<T> = {};
+
+  items.forEach((item: T): void => {
```

Comment:
```suggestion
  items.forEach((item) => {
```

you don't need typing here. TS is already aware what type it is